core/instance.go

```diff
@@ -269,52 +273,222 @@ func (i *instance) getCheapestCompatibleSpotInstanceType(allowedList []string, d
 			i.isAllowed(candidate.instanceType, allowedList, disallowedList) {
 			bestPrice = candidatePrice
 			chosenSpotType = candidate.instanceType
+			cheapest = candidate
 			debug.Println("Best option is now: ", chosenSpotType, " at ", bestPrice)
 		} else if chosenSpotType != "" {
 			debug.Println("Current best option: ", chosenSpotType, " at ", bestPrice)
 		}
 	}
 	if chosenSpotType != "" {
 		debug.Println("Cheapest compatible spot instance found: ", chosenSpotType)
-		return chosenSpotType, nil
+		return cheapest, nil
 	}
-	return chosenSpotType, fmt.Errorf("No cheaper spot instance types could be found")
+	return cheapest, fmt.Errorf("No cheaper spot instance types could be found")
```

Comment:
The first thing I wondered is:I suppose that `cheapest` is valued to `nil` in such case. But I feel that